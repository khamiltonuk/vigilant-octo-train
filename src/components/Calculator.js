import React from "react";
import Keypad from "./Keypad/Keypad";
import Screen from "./Screen/Screen";
import History from "./History/History";

const Calculator = props => {
  const {
    displayValue,
    enterNumber,
    clearInput,
    keypadPress,
    clearAll,
    calculate,
    values,
    removeValue
  } = props;
  return (
    <div className="calculator">
      <Screen
        displayValue={displayValue}
        keypadPress={keypadPress}
        enterNumber={enterNumber}
        wipeInput={clearInput}
      />
      <Keypad
        keypadPress={keypadPress}
        clearInput={clearInput}
        clearAll={clearAll}
        calculate={calculate}
      />
      <History values={values} removeValue={removeValue} />
    </div>
  );
};

export default Calculator;
