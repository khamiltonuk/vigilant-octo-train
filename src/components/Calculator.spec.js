import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import Calculator from "./Calculator";

describe("Calculator", () => {
  test("should render correctly", () => {
    const wrapper = shallow(<Calculator />);
    //console.log(wrapper.debug());
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
