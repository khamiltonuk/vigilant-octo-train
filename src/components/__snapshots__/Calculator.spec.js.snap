// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should render correctly 1`] = `
<div
  className="calculator"
>
  <Screen />
  <Keypad />
  <History />
</div>
`;
